src/test/java/teammates/test/cases/testdriver/TestngTest.java

```diff
@@ -0,0 +1,111 @@
+package teammates.test.cases.testdriver;
+
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.HashMap;
+import java.util.Map.Entry;
+
+import org.testng.annotations.Test;
+
+import teammates.common.util.FileHelper;
+import teammates.test.cases.BaseTestCase;
+
+public class TestngTest extends BaseTestCase {
+
+    @Test
+    public void checkTestsInTestng() throws FileNotFoundException {        
+        String testNgXml = FileHelper.readFile("./src/test/testng.xml");
+        HashMap<String, String> testFiles = getTestFiles(testNgXml, "./src/test/java/teammates/test/cases"); // <class name, package name>
+        
+        testFiles = excludeFilesNotInTestNg(testFiles);
+        
+        for (Entry<String, String> testFileName : testFiles.entrySet()) {
+            assertTrue(isTestFileIncluded(testNgXml, testFileName.getValue(), testFileName.getKey()));
+        }
+    }
+    
+    /**
+     * Files to be checked in testng.xml are added to testFiles
+     * 
+     * @param testNgXml    Contents of testng.xml
+     * @param rootPath     Root path of test files
+     * @return             HashMap containing <class name, package name>
+     */
+    private HashMap<String, String> getTestFiles(String testNgXml, String rootPath) {       
+        // BaseComponentTestCase, BaseTestCase (files in current directory) excluded because 
+        // base classes are extended by the actual tests
+        
+        return addFilesToTestsRecursively(rootPath, true, "", testNgXml); 
```

Comment:
Sorry, one last minor improvement: change the package name passed initially to `teammates.test.cases`. It is more reflective of what actually is happening (the whole string _is_ the package name).